relay/pkg/btc/btc.go

```diff
@@ -40,14 +43,24 @@ type Header struct {
 	Raw []byte
 }
 
-func (h *Header) String() string {
+func (header *Header) Equals(other *Header) bool {
+	if other == nil {
+		return false
+	}
+	return header.Hash == other.Hash &&
+		header.Height == other.Height &&
+		header.PrevHash == other.PrevHash &&
+		header.MerkleRoot == other.MerkleRoot
+}
+
+func (header *Header) String() string {
 	return fmt.Sprintf(
 		"Hash: %s, Height: %d, PrevHash: %s, MerkleRoot: %s, Raw: %s",
-		h.Hash,
-		h.Height,
-		h.PrevHash,
-		h.MerkleRoot,
-		hex.EncodeToString(h.Raw),
+		header.Hash,
+		header.Height,
+		header.PrevHash,
+		header.MerkleRoot,
+		hex.EncodeToString(header.Raw),
 	)
 }
```

Comment:
Just a style note: It's a general Go convention that we use short abbreviations for receiver names. In those cases it should be just `(h *Header)`. If the struct name is more complicated, for example, `BitcoinBlockHash`, we use first letters of all parts: `(bbh *BitcoinBlockHash)`